src/containers/CheckoutPage.jsx

```diff
@@ -527,39 +531,82 @@ class CheckoutPage extends React.Component {
-                  {this.renderShippingOptionsSelector()}
+            <div className="checkout-main-container-wrap">
+              <div className="checkout-main-container">
+                { profileData ? (
+                  <div className="profile-info-container">
+                    <h3 className="profile-info-container-title">
+                      {intl.get('general')}
+                    </h3>
+                    <div className="profile-info-col">
+                      <div className="profile-info-block">
+                        <div className="profile-email-info">
+                          <span className="feedback-label">{ email === '' && intl.get('email-validation') }</span>
+                          <ProfileemailinfoMain profileInfo={profileData} onChange={this.fetchProfileData} />
+                        </div>
+                      </div>
+                    </div>
+                  </div>
+                ) : (<div />)}
+                <div className="profile-info-container">
+                  <h3 className="profile-info-container-title">
+                    {intl.get('shipping')}
+                  </h3>
+                  <div className="profile-info-col">
+                    <div className="profile-info-block">
+                      <div data-region="billingAddressesRegion" style={{ display: 'block' }}>
+                        {this.renderBillingAddressSelector()}
+                      </div>
+                    </div>
+                  </div>
+                  {(needShipmentDetails || deliveries) && (
+                    <div className="profile-info-col">
+                      <div className="profile-info-block">
+                        <div className="checkout-shipping-container">
+                          {this.renderShippingAddressSelector()}
+                          <div data-region="shippingOptionsRegion">
+                            {this.renderShippingOptionsSelector()}
+                          </div>
+                        </div>
+                      </div>
+                    </div>
+                  )}
                 </div>
-              </div>
-              <div data-region="paymentMethodsRegion" style={{ display: 'block' }}>
-                {this.renderPaymentSelector()}
-              </div>
-            </div>
-            <div className="checkout-sidebar" data-region="checkoutOrderRegion" style={{ display: 'block' }}>
-              <div>
-                <div className="checkout-sidebar-inner">
-                  <div data-region="checkoutSummaryRegion" className="checkout-summary-container" style={{ display: 'inline-block' }}>
-                    <CheckoutSummaryList data={orderData} />
+                <div className="profile-info-container">
+                  <h3 className="profile-info-container-title">
+                    {intl.get('payment')}
+                  </h3>
+                  <div className="profile-info-col">
```

Comment:
This block should only be displayed if the gift certificate resource is available